ONSdigital/takeon-business-layer
# spp225: correction in selective editing flag derivation for missing a…

…nd failed flags

# Description



https://collaborate2.ons.gov.uk/jira/browse/SPP-225 Place Selective Editing output (flag) on validation output notification

# How to test?




Env spp225 https://dc-ui-spp225.es-take-on.aws.onsdigital.uk/
Trigger any validation for ref e.g. 12000613746 for periods like 201903 and 201904. Then see logs in lambda - spp-es-takeon-validation-spp225-bpmtransform and check for expected values in selective_editing_flag E.g. 
"Message sent to SNS Topic: {\"context\": {\"log_level\": \"DEBUG\", \"log_correlation_id\": \"33dcbd58-6ef6-435c-9d90-21116560411e\", \"log_correlation_type\": \"Date Adjustment and Validation\", \"survey\": \"023\", \"period\": \"201903\", \"reference\": \"12000613746\"}, \"payload\": {\"reference\": \"12000613746\", \"survey\": \"023\", \"period\": \"201903\", \"selective_editing_flag\": \"FAILED\", \"status\": \"CHECK_NEEDED\", \"error\": null}, \"reference\": \"12000613746\", \"survey\": \"023\", \"period\": \"201903\", \"selective_editing_flag\": \"FAILED\", \"status\": \"CHECK_NEEDED\", \"error\": null}"

# Links


https://github.com/ONSdigital/takeon-validation/pull/61
